### DISCUSSION TITLE
NameError: name 'MyStreamListener' is not defined

### DISCUSSION THREAD

**Discussion by jauntyd** [tweepy/tweepy#1544] [Questions] [answered]
I bet I'm missing something, I just don't know what. Here's the relevant code:

...
myStreamListener = MyStreamListener()
  class MyStreamListener(tweepy.Stream):
    myStream = tweepy.Stream(auth = api.auth, listener=myStreamListener)
...

**NameError: name 'MyStreamListener' is not defined**

Could someone please help me understand this error? :)

**Comment by jauntyd** [comment]
I have pasted the full code in hopes it sheads some light on my troubles.

def tweep(request):
  auth = xxxx
  auth.set_access_token("xxx")
  try:
    redirect_url = auth.get_authorization_url()
  except tweepy.TweepError:
    print('Error! Failed to get request token.')
  api = tweepy.API(auth)
  public_tweets = api.home_timeline()
  for tweet in public_tweets:
    print(tweet.text)
  myStreamListener = MyStreamListener()
  class MyStreamListener(tweepy.Stream):
        #x = 2
    myStream = tweepy.Stream(auth = api.auth, listener=myStreamListener)
  return render(request, "index.html")

**Comment by jauntyd** [comment]
Turns out I wasn't paying enough attention to my console. I needed my auth details included and it worked.

class MyStreamListener(tweepy.Stream):
  def on_status(self, status):
    print(status.text)

Then later....
myStreamListener = MyStreamListener("xxx", "xxx", "xxx", "xxx")

Thanks!

**Comment by Harmon758** [comment, marked as answer]
The initial error was that `MyStreamListener` wasn't defined at all.

For code block usage, see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks.
